modules/apps/object/object-service/src/main/java/com/liferay/object/internal/validation/rule/UniqueCompositeKeyObjectValidationRuleEngineImpl.java

```diff
@@ -130,24 +129,26 @@ private Predicate _getPredicate(
 				continue;
 			}
 
-			String value = null;
+			Object value = entryValues.get(objectField.getName());
 
-			if (StringUtil.equals(
-					objectField.getBusinessType(),
-					ObjectFieldConstants.BUSINESS_TYPE_PICKLIST)) {
+			String stringValue = String.valueOf(value);
+
+			if (value == null) {
```

Comment:
I think it would be better to change this to `Validator.isNull(value)` , no?

Reply:
We can't use that here because we're really only interested in the actual null reference and `Validator.isNull(value)` will return true in other [scenarios](https://github.com/liferay/liferay-portal/blob/master/portal-kernel/src/com/liferay/portal/kernel/util/Validator.java#L911)